docs/docs/60-new-docs/50-user-guide/20-how-to-guides/16-verification.md

```diff
@@ -0,0 +1,215 @@
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: AnalysisRun
+metadata:
+  creationTimestamp: 2025-01-28T20:47:50Z
+  generation: 3
+  labels:
+    kargo.akuity.io/freight-collection: 55d452301040a73e9fd05289b1f8ddbec1791222
+    kargo.akuity.io/promotion: dev.01jjpfgfwkzk18k7cyq61jehf4.319ddec
+    kargo.akuity.io/stage: dev
+  name: dev.01jjqaq7qacfn766tcp1nqz2zv.55d4523
+  namespace: guestbook
+  ownerReferences:
+    - apiVersion: kargo.akuity.io/v1alpha1
+      blockOwnerDeletion: false
+      kind: Freight
+      name: 319ddec6bbfbd808d499a0357b027e773aa3707f
+      uid: fb0db653-92af-420b-9092-0c7bf9be290e
+  resourceVersion: "54013326"
+  uid: 8cb745cb-554b-4141-80ee-e0df181cd56e
+spec:
+  metrics:
+  - name: integration-test
+    provider:
+      job:
+        spec:
+          template:
+            spec:
+              containers:
+              - name: sleep
+                image: alpine:latest
+                command: [sleep, "10"]
+              restartPolicy: Never
+          backoffLimit: 1
+status:
+  dryRunSummary: {}
+  message: Metric "integration-test" assessed Failed due to failed (1) > failureLimit (0)
+  metricResults:
+    - count: 1
+      failed: 1
+      measurements:
+        - finishedAt: 2025-01-28T20:50:31Z
+          metadata:
+            job-name: 8cb745cb-554b-4141-80ee-e0df181cd56e.integration-test.1
+            job-namespace: guestbook
+          phase: Failed
+          startedAt: 2025-01-28T20:47:50Z
+      name: integration-test
+      phase: Failed
+  phase: Failed
+  runSummary:
+    count: 1
+    failed: 1
+  startedAt: 2025-01-28T20:47:50Z
+```
+
+:::note
+The difference between an `AnalysisTemplate` versus an `AnalysisRun` is that an `AnalysisTemplate` is the _definition_ how to perform verification, and the `AnalysisRun` tracks the _progress and result_ of an run of one or more template(s).
+:::
+
+
+##  Arguments and Metadata
+
+It is also possible to specify additional labels, annotations, and arguments
+that should be applied to `AnalysisRun` resources spawned from the referenced
+`AnalysisTemplate`:
+
+```yaml
+apiVersion: kargo.akuity.io/v1alpha1
+kind: Stage
+metadata:
+  name: test
+  namespace: kargo-demo
+spec:
+  # ...
+  verification:
+    analysisTemplates:
+    - name: kargo-demo
+    analysisRunMetadata:
+      labels:
+        foo: bar
+      annotations:
+        bat: baz
+    args:
+    - name: foo
+      value: bar
+```
+
+## Implicit Argo CD Verification
+
+It is usually the desire to wait until a deployment is fully rolled out and updated, before performing any testing and verification. For example, it would be premature to start running tests if a Deployment's Pods were still coming up and or not fully updated to the desired version. Promotions might only include the apply/sync of new manifests to the cluster, but do not not necessarily wait until the rollout was fully complete (which might take minutes depending on many factors like Pod startup time).
```

Comment:
```suggestion
It is usually desirable for changes to be fully rolled out before performing any verification. For example, it would be premature to start running tests if a `Deployment`'s `Pod`s were still starting or not fully transitioned to the desired version. `Promotion`s that integrate with Argo CD do wait for any sync operations they initiate to complete, but it is possible for an Argo CD `Application` to still be `Progressing` even after sync operations have completed and the `Promotion` process is considered complete.
```